cmd/oras/tag/tag.go

```diff
@@ -70,5 +82,22 @@ func tagManifest(opts tagOptions) error {
 		return errors.NewErrInvalidReference(repo.Reference)
 	}
 
-	return oras.Tag(ctx, repo, opts.srcRef, opts.targetRef)
+	rp := &wrapper{
+		repo,
+	}
+
+	return oras.TagN(ctx, rp, opts.srcRef, opts.targetRef, opts.defaultTagNOptions)
+}
+
+type wrapper struct {
+	*remote.Repository
+}
+
+// Override PushReference method to print off which tag(s) were added successfully.
+func (w *wrapper) PushReference(ctx context.Context, expected ocispec.Descriptor, content io.Reader, reference string) error {
+	if err := w.Repository.PushReference(ctx, expected, content, reference); err != nil {
+		return fmt.Errorf("failed %s", reference)
```

Comment:
The `failed %s` error message looks a bit confusing. What does the error message look like if we return the `err` directly?

Reply:
the error itself looks like: `Error: failed to tag localhost:12900/hello:latest as tag55:1,: invalid reference: invalid tag`
changed to what Shiwei suggested `fmt.Errorf("%s: %w", reference, err)` and now error looks like: `Error: failed to tag localhost:12900/hello:latest as tag55:1,: tag55:1,: invalid reference: invalid tag`